actors/builtin/miner/miner_actor.go

```diff
@@ -1998,9 +2030,23 @@ func validateReplaceSector(rt Runtime, st *State, store adt.Store, params *PreCo
 	if len(replaceSector.DealIDs) > 0 {
 		rt.Abortf(exitcode.ErrIllegalArgument, "cannot replace sector %v which has deals", params.ReplaceSectorNumber)
 	}
-	if params.SealProof != replaceSector.SealProof {
-		rt.Abortf(exitcode.ErrIllegalArgument, "cannot replace sector %v seal proof %v with seal proof %v",
-			params.ReplaceSectorNumber, replaceSector.SealProof, params.SealProof)
+	if nv < network.Version7 {
+		if params.SealProof != replaceSector.SealProof {
+			rt.Abortf(exitcode.ErrIllegalArgument, "cannot replace sector %v seal proof %v with seal proof %v",
+				params.ReplaceSectorNumber, replaceSector.SealProof, params.SealProof)
+		}
+	} else {
+		// From network version 7, the new sector's seal type must have the same Window PoSt proof type as the one
+		// being replaced, rather than be exactly the same seal type.
+		// This permits replacing sectors with V1 seal types with V1_1 seal types.
+		replaceWPoStProof, err := replaceSector.SealProof.RegisteredWindowPoStProof()
```

Comment:
Consider extracting this logic since it's duplicated at least once.

Reply:
I'm going to do this one

Reply:
After poking around a bit I decided not to do this.  Might be worth coming back to when we clear away old network version checks in v3.